Ray-Only.ipynb

```diff
@@ -46,70 +86,459 @@
+    "2. Union of dataframes is supported, can be done before the compaction step. https://github.com/ray-project/ray/issues/24598\n",
+    "Status: Fails due to memory limits, cannot load a dataframe for more than several days worth of parquets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "source_hidden": true
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Read path\n",
+    "read_path = f'{prefixPathRead}/metric={metricName}'\n",
+    "read_bucket_uri = f's3://{bucketName}/{read_path}/year={year}/month={month}/day={day}'\n",
+    "# Holds paths \n",
+    "read_bucket_multiple = []\n",
+    "\n",
+    "# Creating paths for every day in the month.\n",
+    "for i in range(2,32):\n",
+    "    read_bucket_multiple.append(f\"s3://{bucketName}/{read_path}/year={year}/month={month}/day=\"f\"{i:02d}\")\n",
+    "                           \n",
+    "#print(read_bucket_multiple)\n",
+    "\n",
+    "\n",
+    "# Write path\n",
+    "write_path = f\"{prefixPathWrite}/metric={metricName}\"\n",
+    "write_bucket_uri = f's3://{bucketName}/{write_path}/year={year}/month={month}'\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "source_hidden": true
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# S3FileSystem in PyArrow\n",
+    "fs_pyarrow = pq.S3FileSystem(endpoint_override=endpoint)\n",
+    "\n",
+    "\n",
+    "# Initial read - First day to start a dataframe\n",
+    "master_df = ray.data.read_parquet(paths=f's3://{bucketName}/{prefixPathRead}/metric={metricName}/year={year}/month={month}/day=01', filesystem=fs_pyarrow)\n",
+    "\n",
+    "# Reading remaining days\n",
+    "for x in read_bucket_multiple:\n",
+    "    df = ray.data.read_parquet(paths=x, filesystem=fs_pyarrow) # Reading parquets for each day using Ray through filesystem\n",
+    "    master_df = master_df.union(df) # Followed by unioning the dataframe each time\n",
+    "    \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "source_hidden": true
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Writing back a single parquet file\n",
+    "df.repartition(1).write_parquet(path=write_bucket_uri, filesystem=fs_pyarrow)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Making use of datetime to run compaction for every day, union dataframes and finally writeback"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "source_hidden": true
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "firstDay = date(2021, 1, 1)\n",
+    "lastDay = firstDay + relativedelta(months = 1)\n",
+    "print(lastDay, firstDay)\n",
```

Comment:
there are a lot of these print statements with no context, which may be useful for debugging but is hard to decipher for other users who are just reading result cells' messages.  Could you either add some accompanying text to define what these messages are for, or remove them if they were only used for development/debugging?

Reply:
Done!